diff --git a/build/src/export/exportHermes.js b/build/src/export/exportHermes.js
index 9068adbcad970e9b1414aa2c41183462d76da584..2a16f616445afd5d39b61397cfc6ac7ce28deedc 100644
--- a/build/src/export/exportHermes.js
+++ b/build/src/export/exportHermes.js
@@ -100,12 +100,14 @@ async function assertEngineMismatchAsync(projectRoot, exp, platform) {
     await maybeThrowFromInconsistentEngineAsync(projectRoot, configFilePath, platform, isHermesManaged);
 }
 function isEnableHermesManaged(expoConfig, platform) {
+    var _expoConfig_extra;
+    const configuredEngine = (_expoConfig_extra = expoConfig.extra) == null ? void 0 : _expoConfig_extra.nemuJsEngine;
     switch(platform){
         case 'android':
             {
                 var _expoConfig_android;
                 // NOTE(@kitten): `jsEngine` was deprecated, but we're preserving the check
-                return (((_expoConfig_android = expoConfig.android) == null ? void 0 : _expoConfig_android.jsEngine) ?? expoConfig.jsEngine) !== 'jsc';
+                return (((_expoConfig_android = expoConfig.android) == null ? void 0 : _expoConfig_android.jsEngine) ?? expoConfig.jsEngine ?? configuredEngine) !== 'jsc';
             }
         case 'ios':
         case 'tvos':
@@ -113,7 +115,7 @@ function isEnableHermesManaged(expoConfig, platform) {
             {
                 var _expoConfig_ios;
                 // NOTE(@kitten): `jsEngine` was deprecated, but we're preserving the check
-                return (((_expoConfig_ios = expoConfig.ios) == null ? void 0 : _expoConfig_ios.jsEngine) ?? expoConfig.jsEngine) !== 'jsc';
+                return (((_expoConfig_ios = expoConfig.ios) == null ? void 0 : _expoConfig_ios.jsEngine) ?? expoConfig.jsEngine ?? configuredEngine) !== 'jsc';
             }
         default:
             return false;
