diff --git a/build/src/export/exportHermes.js b/build/src/export/exportHermes.js
index 155f189..ae3d168 100644
--- a/build/src/export/exportHermes.js
+++ b/build/src/export/exportHermes.js
@@ -113,18 +113,19 @@ async function assertEngineMismatchAsync(projectRoot, exp, platform) {
     await maybeThrowFromInconsistentEngineAsync(projectRoot, configFilePath, platform, isHermesManaged);
 }
 function isEnableHermesManaged(expoConfig, platform) {
+    const configuredEngine = expoConfig.extra?.nemuJsEngine;
     switch(platform){
         case 'android':
             {
                 // NOTE(@kitten): `jsEngine` was deprecated, but we're preserving the check
-                return (expoConfig.android?.jsEngine ?? expoConfig.jsEngine) !== 'jsc';
+                return (expoConfig.android?.jsEngine ?? expoConfig.jsEngine ?? configuredEngine) !== 'jsc';
             }
         case 'ios':
         case 'tvos':
         case 'macos':
             {
                 // NOTE(@kitten): `jsEngine` was deprecated, but we're preserving the check
-                return (expoConfig.ios?.jsEngine ?? expoConfig.jsEngine) !== 'jsc';
+                return (expoConfig.ios?.jsEngine ?? expoConfig.jsEngine ?? configuredEngine) !== 'jsc';
             }
         default:
             return false;
